openmdao/test/util.py

```diff
@@ -5,6 +5,40 @@
 from math import isnan
 import numpy as np
 
+
+
+def problem_derivatives_test(unittest, problem, tol = 1e-5):
+    """Runs partial derivates check on an OpenMDAO problem instance.
```

Comment:
Excellent. Now, it just needs a test. You can piggyback it onto an existing test somewhere, maybe in the check_partials test.